src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-search.component.ts

```diff
@@ -741,6 +750,71 @@ export class GermplasmSearchComponent implements OnInit {
         const confirmModalRef = this.modalService.open(KeySequenceRegisterDeletionDialogComponent as Component);
         confirmModalRef.componentInstance.gids = gids;
     }
+
+    groupGermplasm() {
+        if (!this.validateSelection()) {
+            return;
+        }
+        if (this.isSelectAll) {
+            this.alertService.error('germplasm-grouping.group-all-germplasm-not-supported');
+            return;
+        }
+        if (this.size(this.selectedItems) > 500) {
+            this.alertService.error('germplasm-grouping.too-many-selected-germplasm');
+            return;
+        }
+        const groupGermplasmModal = this.modalService.open(GermplasmGroupOptionsDialogComponent as Component);
+        groupGermplasmModal.componentInstance.gids = this.getSelectedItemIds();
+        groupGermplasmModal.result.then(((germplasmGroupList) => {
+            this.openGermplasmGroupingResult(germplasmGroupList)
+        }));
+    }
+
+    openGermplasmGroupingResult(results) {
+        if (results) {
+            const germplasmGroupingResultComponent = this.modalService.open(GermplasmGroupingResultComponent as Component, { size: 'lg', backdrop: 'static' });
+            germplasmGroupingResultComponent.componentInstance.results = results;
+            germplasmGroupingResultComponent.result.then(() => {
+                    this.hiddenColumns[ColumnLabels['GROUP ID']] = false;
+                    this.resetTable();
```

Comment:
Can you use this.transition() so that it will just refresh the current germplasm search page.